launch/pipeline/service.py

```diff
@@ -45,10 +65,26 @@ def _init_callable_service(self):
             ), "`kwargs` is given, but the service is not a class"
             self._callable_service = self.service
 
+    def set_make_request(
+        self, make_request: Callable, force: bool = False
+    ) -> None:
+        if force:
```

Comment:
What is this doing? Why would a user call `set_make_request` with `force=False`? Why would they not set this in `__init__` ?

Reply:
`set_make_request` is an internal backend functionality to "override" call function. The backend gives a function `make_request` that in turn is able to call for example celery queue associated with that service.

The trick here was the fact that we need to apply `set_make_request` recursively to all of the subservices. The higher level the parent, if asked for `call` need to execute the user code themselves. All of the sub-services by default are calling http/celery endpoints.
We want set_make_request to take effect only for lower level services, not the root. Otherwise the'll be the deadlock: Service A receives a request and uses `make_request` to `call` service A, and so on.